descqa/DeltaSigmaTest.py

```diff
@@ -0,0 +1,118 @@
+import os
+import numpy as np
+from scipy.interpolate import interp1d
+from astropy import units as u
+from astropy.coordinates import SkyCoord, search_around_sky
+import astropy.constants as cst
+from astropy.cosmology import WMAP7 # pylint: disable=no-name-in-module
+from .base import BaseValidationTest, TestResult
+from .plotting import plt
+
+__all__ = ['DeltaSigmaTest']
+
+class DeltaSigmaTest(BaseValidationTest):
+    """
+    This validation test looks at galaxy-shear correlations by comparing Delta
+    Sigma to the Singh et al (2015) measurements on the SDSS LOWZ sample.
+    """
+
+    def __init__(self, **kwargs):
+        # pylint: disable=super-init-not-called
+
+        # validation data
+        validation_filepath = os.path.join(self.data_dir, kwargs['data_filename'])
+        zmax = kwargs['zmax']
+
+        self.validation_data = np.loadtxt(validation_filepath)
+
+        # Create interpolation tables for efficient computation of sigma crit
+        z = np.linspace(0, zmax, zmax*100)
+        d1 = WMAP7.angular_diameter_distance(z) # in Mpc
+        self.angular_diameter_distance = interp1d(z, d1, kind='quadratic')
+
+        d2 = WMAP7.comoving_transverse_distance(z) # in Mpc
+        self.comoving_transverse_distance = interp1d(z, d2, kind='quadratic')
+
+    def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
+        # pylint: disable=no-member
+
+        res = catalog_instance.get_quantities(['redshift_true', 'ra', 'dec', 'shear_1', 'shear_2',
+                                               'convergence', 'mag_true_i_sdss', 'mag_true_z_sdss',
+                                               'mag_true_g_sdss', 'mag_true_r_sdss'])
+
+        # Compute mask for lowz sample
+        cperp = (res['mag_true_r_sdss'] - res['mag_true_i_sdss']) - (res['mag_true_g_sdss'] - res['mag_true_r_sdss'])/4.0 - 0.18
+        cpar = 0.7*(res['mag_true_g_sdss'] - res['mag_true_r_sdss']) + 1.2*((res['mag_true_r_sdss'] - res['mag_true_i_sdss'])-0.18)
+
+        mask_lowz = np.abs(cperp) < 0.2 # color boundaries
+        mask_lowz &= res['mag_true_r_sdss'] < (13.5 + cpar/0.3) # sliding magnitude cut
+        mask_lowz &= (res['mag_true_r_sdss'] > 16) &(res['mag_true_r_sdss'] < 19.6)
+
+        # Counting the number density of LOWZ galaxies
+        nlens = len(np.where(mask_lowz)[0]) / catalog_instance.sky_area
```

Comment:
@EiffL @rmandelb - should this `nlens` count be done here, or after the subsequent 0.16 < z < 0.36 redshift cut on line 57 below?

Reply:
The code is right - it's before.  The numbers in the target-selection paper do not involve a redshift cut (since it's the target density before spectra are taken).  So Francois had to calculate the LOWZ density before the redshift cut.  But since Sukhdeep's lensing profiles involved a redshift cut, Francois then made the redshift cut before comparing lensing profiles.

Reply:
Ok, just checking, thanks for clarifying

Reply:
It's good you asked - I think that this should be documented with a comment so that when someone else is trying to update this test a few years from now when running on some other catalogs, they know why it's like this!  Seemingly arcane choices should be documented for the sake of future code writers/maintainers  :)

Reply:
Even better, it would probably be useful to peel apart this into a standalone CMASS selection function, since that is also being used elsewhere to check number densities. I now have my own implementation for model recalibration purposes, so this is no rush so far as I'm concerned, and it's also fine if we stick with the principle of one, two, infinity for now. But the docstring to such a function would be the natural place to document something that's getting reused.